### PR TITLE
Added support for gp3 and io2 both root and ebs volumes (fixed copy of #117)

### PR DESCRIPTION
**Pull request by tbpoetke**
## what
- Added support for io2 and gp3 volumes

## why
- original PR had conflicts, this will work hopefully
- io2 and gp3 are new more performant volumes therefore they should be supported

## references
- https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#ebs-ephemeral-and-root-block-devices
- Closes https://github.com/cloudposse/terraform-aws-ec2-instance/issues/114

### CODE REVIEW

**Comment by nitrocode** [comment]
@tbpoetke please bump modules in the examples in order to successfully pass the test/bats check

https://github.com/tbpoetke/terraform-aws-ec2-instance/blob/master/examples/complete/main.tf

**Comment by nitrocode** [comment]
@tbpoetke unfortunately after bumping the modules, there are also some breaking changes with the interface. Please ensure that the examples/complete module plans locally successfully and I will re-run tests.

**Comment by tbpoetke** [comment]
> @tbpoetke unfortunately after bumping the modules, there are also some breaking changes with the interface. Please ensure that the examples/complete module plans locally successfully and I will re-run tests.

ok, I will try to fix it, thanks for your help!

**Comment by tbpoetke** [comment]
@nitrocode should work now, terraform validate and terraform plan no more errors, please retry the Github Action test. Thnak you!

**Comment by tbpoetke** [comment]
Ok, will fix it again tomorrow ....sorry for the circumstances ...

**Review by tbpoetke — COMMENTED** [review]
I changed the deprecated syntax. Locally it's working for me, idk why the tests are failing!

**Comment by nitrocode** [comment]
@tbpoetke more yak shaving... please bump the required terraform version to `> 1.0` in `versions.tf` and `examples/complete/versions.tf`

**Comment by tbpoetke** [comment]
It's a long ride through this pipeline :1234:

File: examples/complete/main.tf [hunk 1]
```diff
@@ -1,6 +1,3 @@
-provider "aws" {
```
Comment (nitrocode) [inline examples/complete/main.tf:1]: Why is this removed?

File: examples/complete/provider.tf [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+provider "aws" {
+  region = "us-east-2"
+  assume_role {
+    role_arn = "arn:aws:iam::128840427886:role/OrganizationAccountAccessRole"
+  }
+}
```
Comment (nitrocode) [inline examples/complete/provider.tf:7]: Please move this block back to `main.tf` and remove the `role_arn` assumption

Comment (tbpoetke) [reply examples/complete/provider.tf:7]: sry, accidentally checked in my test setup ...

**Review by nitrocode — APPROVED** [review]

**Comment by nitrocode** [comment]
Thank you so much for the contribution @tbpoetke ! I know it was a bit arduous. Future PRs will be easier.

**Comment by tbpoetke** [comment]
Thank you for your review and your help @nitrocode . Next time I will try to test the pipeline with arc to avoid such problems ....
